esphome/components/font/__init__.py

```diff
@@ -161,26 +200,45 @@ def download_gfonts(value):
         )
 
     ttf_url = match.group(1)
-    try:
-        req = requests.get(ttf_url, timeout=30)
-        req.raise_for_status()
-    except requests.exceptions.RequestException as e:
-        raise cv.Invalid(f"Could not download ttf file for {name} ({ttf_url}): {e}")
+    _LOGGER.debug("download_gfont: ttf_url=%s", ttf_url)
 
-    path.parent.mkdir(exist_ok=True, parents=True)
-    path.write_bytes(req.content)
+    download_content(ttf_url, path)
     return value
 
 
-GFONTS_SCHEMA = cv.All(
+def download_web_font(value):
+    url = value[CONF_URL]
+    path = get_font_path(value)
+
+    download_content(url, path)
+    _LOGGER.debug("download_web_font: path=%s", path)
+    return value
+
+
+EXTERNAL_FONT_SCHEMA = cv.Schema(
     {
-        cv.Required(CONF_FAMILY): cv.string_strict,
         cv.Optional(CONF_WEIGHT, default="regular"): cv.Any(
             cv.int_, validate_weight_name
         ),
         cv.Optional(CONF_ITALIC, default=False): cv.boolean,
-    },
-    download_gfonts,
+        cv.Optional(CONF_REFRESH, default="1d"): cv.All(cv.string, cv.source_refresh),
+    }
+)
+
+CONF_DOWNLOAD = "download"
+
+GFONTS_SCHEMA = EXTERNAL_FONT_SCHEMA.extend(
+    {
+        cv.Required(CONF_FAMILY): cv.string_strict,
+        cv.Optional(CONF_DOWNLOAD): download_gfont,
+    }
+).extend(EXTERNAL_FONT_SCHEMA)
+
+WEB_FONT_SCHEMA = EXTERNAL_FONT_SCHEMA.extend(
+    {
+        cv.Required(CONF_URL): cv.string_strict,
+        cv.Optional(CONF_DOWNLOAD): download_web_font,
+    }
 )
```

Comment:
No, it should be like this I think.
```suggestion
GFONTS_SCHEMA = cv.All(
    EXTERNAL_FONT_SCHEMA.extend(
        {
            cv.Required(CONF_FAMILY): cv.string_strict,
        }
    ),
    download_gfont,
)

WEB_FONT_SCHEMA = cv.All(
    EXTERNAL_FONT_SCHEMA.extend(
        {
            cv.Required(CONF_URL): cv.string_strict,
        }
    ),
    download_web_font,
)
```

Reply:
nice I couldn't figure out how to get the cv.All to have a schema. Updated, hopefully it looks good!